pulp_ansible/tests/functional/cli/test_collection_download_count.py

```diff
@@ -0,0 +1,109 @@
+"""Tests that Collections hosted by Pulp can be installed by ansible-galaxy."""
+import subprocess
+import pytest
+from pulpcore.client.pulp_ansible import AnsibleRepositorySyncURL
+
+from pulp_ansible.tests.functional.constants import (
+    ANSIBLE_DEMO_COLLECTION,
+    ANSIBLE_DEMO_COLLECTION_VERSION,
+    GALAXY_ANSIBLE_BASE_URL,
+)
+from pulp_ansible.tests.functional.utils import gen_ansible_remote
+
+
+@pytest.fixture
+def install_scenario_distribution(
+    delete_orphans_pre,
+    ansible_repo_api_client,
+    ansible_repo_factory,
+    ansible_collection_remote_factory,
+    ansible_distribution_factory,
+    monitor_task,
+):
+    """Prepare a distribution to install from."""
+    collection_requirements = f"collections:\n  - {ANSIBLE_DEMO_COLLECTION}"
+    remote = ansible_collection_remote_factory(
+        **gen_ansible_remote(url=GALAXY_ANSIBLE_BASE_URL, requirements_file=collection_requirements)
+    )
+    repo = ansible_repo_factory(remote=remote.pulp_href)
+
+    # Sync
+    repository_sync_data = AnsibleRepositorySyncURL()
+    sync_response = ansible_repo_api_client.sync(repo.pulp_href, repository_sync_data)
+    monitor_task(sync_response.task)
+
+    # Distribute
+    return ansible_distribution_factory(repo)
+
+
+def test_collection_download_count(
+    ansible_dir_factory,
+    galaxy_v3_collections_api_client,
+    install_scenario_distribution,
+    pulp_admin_user,
+    pulp_settings,
+):
+    """Test that the collection download counting functions."""
+    if not pulp_settings.ANSIBLE_COLLECT_DOWNLOAD_COUNT:
+        pytest.skip("ANSIBLE_COLLECT_DOWNLOAD_COUNT not enabled")
+    with pulp_admin_user:
+        collection_namespace = ANSIBLE_DEMO_COLLECTION.split(".")[0]
+        collection_name = ANSIBLE_DEMO_COLLECTION.split(".")[1]
+        collection_detail = galaxy_v3_collections_api_client.read(
+            path=install_scenario_distribution.base_path,
+            namespace=collection_namespace,
+            name=collection_name,
+        )
+
+        collection_detail = galaxy_v3_collections_api_client.read(
+            path=install_scenario_distribution.base_path,
+            namespace=collection_namespace,
+            name=collection_name,
+        )
```

Comment:
```suggestion
```
This call really looks redundant to me. If you say, the dl-count is interfered with by another test, let's not check that it goes from 2 to 3, but that it increases by exactly 1.

Reply:
I'm going to re-write this to be clearer.